### PR TITLE
Sales

### PR DESCRIPTION
**Pull request by azizchelchel**

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -11,64 +11,93 @@ datasource db {
...
-  isDeleted     Boolean  @default(false)
-}
\ No newline at end of file
+  price         Int     @default(0)
+  isDeleted     Boolean @default(false)
+  item          Item[]
+}
+
+model Item {
+  itemId    Int   @id @unique @default(autoincrement())
```
Comment (Salem-Abderaouf) [inline prisma/schema.prisma:86]: for best practice use `id` instead of itemid

File: prisma/schema.prisma [hunk 2]
```diff
@@ -11,64 +11,93 @@ datasource db {
...
+  drugId    Int
+  drugPrice Float @default(0)
+  quantity  Int
+  totalItem Float @default(0)
```
Comment (Salem-Abderaouf) [inline prisma/schema.prisma:90]: use more descriptive name,
NB: for totalItem you already have this information you don't need to repeat it

File: prisma/schema.prisma [hunk 3]
```diff
@@ -11,64 +11,93 @@ datasource db {
...
+  drug      Drug  @relation(fields: [drugId], references: [drug_id])
+  sales     Sales @relation(fields: [saleId], references: [saleId])
+  saleId    Int
+}
+
+model Sales {
+  saleId       Int      @id @default(autoincrement())
```
Comment (Salem-Abderaouf) [inline prisma/schema.prisma:97]: same use id

File: models/sales.model.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+export const addSaleInDb = (data) => {
+    const {userId, items} = data;
+    // calculate totalItem 
+    items.map(e => e.totalItem = parseInt(e.drugPrice) * parseInt(e.quantity)
+    );
+    // calculate totalSale
+    let totalSale = 0;
+    items.forEach(e => totalSale = totalSale + e.totalItem);
+
+    return new Promise(
```
Comment (Salem-Abderaouf) [inline models/sales.model.js:15]: Remove the `new Promise` and `(resolve,reject)` 
Instead use 
`export const addSaleInDb = async(data)=>{
// your_code 
await prisma.sales.create(
// your_code
)
}`
async and await are the simplicity of the promise in ES6 to make The code more readable

File: models/sales.model.js [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+            )  
+        }
+    )
+};
+
+
+export const getSalesByDateFromDb = (purchaseDate) => {
+    return new Promise(
+        async (resolve,reject) => {
+            const toDay = new Date(purchaseDate); //purchcacseDate without time part
```
Comment (Salem-Abderaouf) [inline models/sales.model.js:115]: rename the toDay to day or purchaseDate

File: models/sales.model.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+            const dayAfter = new Date(date1.getTime());
```
Comment (Salem-Abderaouf) [inline models/sales.model.js:116]: date1 is undefined

**Review by Salem-Abderaouf — CHANGES_REQUESTED** [review]

**Review by Aminedifdz — COMMENTED** [review]
Reviewed.

File: models/sales.model.js (hunk 4 above)
Comment (azizchelchel) [reply models/sales.model.js:15]: ok thanks i'l redo the work as you suggest

**Pull request merged**